crescent/commands/hooks.py

```diff
@@ -42,9 +43,31 @@ def hook(
         return partial(hook, callback, after)  # type: ignore
     if not iscoroutinefunction(callback):
         raise ValueError(f"Function `{callback.__name__}` must be async.")
-    if after:
-        command.metadata.after_hooks.insert(0, callback)
-    else:
-        command.metadata.hooks.insert(0, callback)
+    command.metadata.add_hooks([callback], prepend=True, after=after)
 
     return command
+
+
+class HasHooks(Protocol):
+    hooks: list[HookCallbackT] | None
+    after_hooks: list[HookCallbackT] | None
+
+
+class HasHooksLongName(Protocol):
+    command_hooks: list[HookCallbackT] | None
+    command_after_hooks: list[HookCallbackT] | None
+
+
+def add_hooks(obj: HasHooks | HasHooksLongName, command: Includable[Any]) -> None:
+    if not isinstance(command.metadata, AppCommandMeta):
+        return
+
+    command_hooks = getattr(obj, "hooks", None) or getattr(obj, "command_hooks", None)
+    command_after_hooks = getattr(obj, "after_hooks", None) or getattr(
+        obj, "command_after_hooks", None
+    )
```

Comment:
I don't like this. Would it be possible to just change everything to use `command_hooks` and `command_after_hooks`?

Reply:
This is a breaking change because you do `hooks=[]` when making a group and `command_hooks=[]` when making a bot/plugin.